nixos/modules/services/web-apps/hedgedoc.nix

```diff
@@ -1018,58 +271,99 @@ in
         `HedgeDoc` is running.
       '';
     };
-
-    package = mkOption {
-      type = types.package;
-      default = pkgs.hedgedoc;
-      defaultText = literalExpression "pkgs.hedgedoc";
-      description = lib.mdDoc ''
-        Package that provides HedgeDoc.
-      '';
-    };
-
   };
 
   config = mkIf cfg.enable {
-    assertions = [
-      { assertion = cfg.settings.db == {} -> (
-          cfg.settings.dbURL != "" && cfg.settings.dbURL != null
-        );
-        message = "Database configuration for HedgeDoc missing."; }
-    ];
-    users.groups.${name} = {};
+    users.groups.${name} = { };
     users.users.${name} = {
       description = "HedgeDoc service user";
       group = name;
-      extraGroups = cfg.groups;
-      home = cfg.workDir;
-      createHome = true;
       isSystemUser = true;
     };
 
-    systemd.services.hedgedoc = {
-      description = "HedgeDoc Service";
-      wantedBy = [ "multi-user.target" ];
-      after = [ "networking.target" ];
-      preStart = ''
-        ${pkgs.envsubst}/bin/envsubst \
-          -o ${cfg.workDir}/config.json \
-          -i ${prettyJSON cfg.settings}
-        mkdir -p ${cfg.settings.uploadsPath}
-      '';
-      serviceConfig = {
-        WorkingDirectory = cfg.workDir;
-        StateDirectory = [ cfg.workDir cfg.settings.uploadsPath ];
-        ExecStart = "${cfg.package}/bin/hedgedoc";
-        EnvironmentFile = mkIf (cfg.environmentFile != null) [ cfg.environmentFile ];
-        Environment = [
-          "CMD_CONFIG_FILE=${cfg.workDir}/config.json"
-          "NODE_ENV=production"
-        ];
-        Restart = "always";
-        User = name;
-        PrivateTmp = true;
-      };
+    services.hedgedoc.settings = {
+      defaultNotePath = mkDefault "${cfg.package}/public/default.md";
+      docsPath = mkDefault "${cfg.package}/public/docs";
+      viewPath = mkDefault "${cfg.package}/public/views";
     };
+
+    systemd.services.hedgedoc =
+      let
+        # Some settings that we set as null will break hedgedoc,
+        # as it expects them either to be undefined or set.
+        cleanConfig = pkgs.runCommandLocal "hedgedoc-config.json"
+          {
+            nativeBuildInputs = [ pkgs.jq ];
+          } ''
+          jq '{production:del(.[]|nulls)|del(.[][]?|nulls)}' \
+            < ${settingsFormat.generate "hedgedoc-ugly.json" cfg.settings} \
+            > $out
+        '';
+      in
+      {
+        description = "HedgeDoc Service";
+        documentation = [ "https://docs.hedgedoc.org/" ];
+        wantedBy = [ "multi-user.target" ];
+        after = [ "networking.target" ];
+        preStart = ''
+          ${pkgs.envsubst}/bin/envsubst \
+            -o /var/lib/${name}/config.json \
+            -i ${cleanConfig}
+          ${pkgs.coreutils}/bin/mkdir -p ${cfg.settings.uploadsPath}
```

Comment:
Please use tmpfiles.d for this

Reply:
Didn't know about tempfiles.d. Thanks!

Reply:
I ended up putting the config in `/run/hedgedoc`, since it isn't really permanent state.